src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -606,13 +606,54 @@ internal static bool TryGetDebuggerDisplayInfo(this DkmClrValue value, out Debug
             return displayInfo.HasValues;
         }
 
+        /// <summary>
+        /// Custom comparer for the GetDebuggerCustomUIVisualizerInfo method so that our custom set can
+        /// eliminate duplicated visualizer attributes from a type due to inheritance.
+        /// </summary>
+        private class DkmClrDebuggerVisualizerAttributeComparer : IEqualityComparer<DkmClrDebuggerVisualizerAttribute>
+        {
+            public bool Equals(DkmClrDebuggerVisualizerAttribute x, DkmClrDebuggerVisualizerAttribute y)
+            {
+                if (ReferenceEquals(x, y))
+                {
+                    return true;
+                }
+
+                return x.VisualizerDescription == y.VisualizerDescription &&
```

Comment:
>x.VisualizerDescription == y.VisualizerDescription [](http://example.com/codeflow?start=23&length=50)

Is it valid for `GetDebuggerCustomUIVisualizerInfo()` to return two attributes for the same type with distinct description?